client/components/data/query-publicize-connections/index.jsx

```diff
@@ -34,24 +34,29 @@ class QueryPublicizeConnections extends Component {
 }
 
 QueryPublicizeConnections.propTypes = {
-	siteId: PropTypes.number,
+	requestConnections: PropTypes.func,
 	requestingConnections: PropTypes.bool,
-	requestConnections: PropTypes.func
+	selectedSite: PropTypes.bool,
+	siteId: PropTypes.number,
 };
 
 QueryPublicizeConnections.defaultProps = {
-	requestConnections: () => {}
+	requestConnections: () => {},
+	requestingConnections: false,
+	selectedSite: false,
+	siteId: 0,
 };
 
 export default connect(
-	( state, ownProps ) => {
+	( state, { siteId, selectedSite } ) => {
+		siteId = siteId || ( selectedSite && getSelectedSiteId( state ) );
```

Comment:
How about the inverse? If someone provides both args we'll always just fetch the selectedSite.
```jsx
const selectedSiteId = selectedSite && getSelectedSiteId( state );
const siteId = selectedSiteId || siteId;
```